### PR TITLE
feat: finish working at task3

### PR DESCRIPTION
**Pull request by dmytromykhailiuk**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,37 +1,16 @@
...
+    <React.StrictMode>
+      <ErrorBoundary>
+        <MainPage />
+        <Footer />
+      </ErrorBoundary>
+    </React.StrictMode>
   );
 }
+
+export default App;
```
Comment (StadnykYura) [inline src/App.tsx:16]: set newline at the end of each file.

File: src/pages/main/MainPage.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React, { useState } from "react";
+import { Banner, MovieBoard } from "./components";
+import { movies as allMovies } from "../../mocks"
+import { filterMovies, sortMovies } from "../../shared/helpers";
+
+const MainPage: React.FC = () => {
+  const [categories, onChangeCategories] = useState("ALL");
```
Comment (StadnykYura) [inline src/pages/main/MainPage.tsx]: make 'ALL' - enum or constant for application

Comment (StadnykYura) [inline src/pages/main/MainPage.tsx]: why plural?

File: src/pages/main/MainPage.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  const [searchingValue, onChangeSearchingValue] = useState("");
```
Comment (StadnykYura) [inline src/pages/main/MainPage.tsx]: common practice is to write 
`[searchingValue, setSearchingValue]`
use this approach to other `useState` vallues

File: src/pages/main/components/Banner/Banner.module.scss [hunk 4]
```diff
@@ -0,0 +1,63 @@
+@import '../../../../scss/mixins';
+@import '../../../../scss/variables';
```
Comment (StadnykYura) [inline src/pages/main/components/Banner/Banner.module.scss]: it looks like the folder is a good candidate for alias

File: src/pages/main/components/CategoryPanel/CategoryPanel.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import React, { useCallback, useState } from "react";
+import { categories } from "../../../../mocks";
+import classes from "./CategoryPanel.module.scss";
+
+interface CategoryPanelProps {
+  onChangeCategories?: (category: string) => void;
+}
+
+const ALL = 'ALL';
```
Comment (StadnykYura) [inline src/pages/main/components/CategoryPanel/CategoryPanel.tsx]: used in few files, create constants file, and export it

File: src/pages/main/components/CategoryPanel/CategoryPanel.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+const CategoryPanel: React.FC<CategoryPanelProps> = ({onChangeCategories = () => {}}) => {
+  const [selectedCategory, selectCategory] = useState("ALL");
```
Comment (StadnykYura) [inline src/pages/main/components/CategoryPanel/CategoryPanel.tsx]: string in `useState' but you have already created constant

File: src/pages/main/components/CategoryPanel/CategoryPanel.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+  const onSelectCategory = useCallback((selectedCategory: string) => {
+    selectCategory(selectedCategory);
+    onChangeCategories(selectedCategory);
+  }, []);
+
+  return (
+    <ul className={classes['category-panel']}>
```
Comment (StadnykYura) [inline src/pages/main/components/CategoryPanel/CategoryPanel.tsx:14]: why did you change the way u use classes?

File: src/pages/main/components/CategoryPanel/CategoryPanel.tsx (hunk 6 above)
Comment (StadnykYura) [inline src/pages/main/components/CategoryPanel/CategoryPanel.tsx]: why do you need `selectedCategory` here, if you have already created state in your parent for this
```const [categories, onChangeCategories] = useState("ALL"); ```

File: src/pages/main/components/MovieBoard/MovieBoard.tsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  onChangeCategories?: (categories: string) => void;
+  onChangeSortingDirection?: (isDownDirection: boolean) => void;
+  movies?: Movie[]
+}
+
+const MovieBoard: React.FC<MovieBoardProps> = ({ 
+  movies = [],
+  onChangeCategories = () => {}, 
+  onChangeSortingDirection = () => {}
+}) => {
```
Comment (StadnykYura) [inline src/pages/main/components/MovieBoard/MovieBoard.tsx]: why do you need default values and why these all props here are optional?

File: src/pages/main/components/MovieBoard/MovieBoard.tsx [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+  return (
+    <div className={classes['movie-board']}>
```
Comment (StadnykYura) [inline src/pages/main/components/MovieBoard/MovieBoard.tsx:21]: please use one approach for className

File: src/pages/main/components/SortPanel/SortPanel.tsx [hunk 10]
```diff
@@ -0,0 +1,34 @@
+import React, { useCallback, useState } from 'react';
+import classes from './SortPanel.module.scss';
+
+interface SortPanelProps {
+  onChangeSortingDirection?: (isDownDirection: boolean) => void;
+}
+
+const SortPanel: React.FC<SortPanelProps> = ({ onChangeSortingDirection = () => {} }) => {
+  const [isDown, changeDirection] = useState(true);
```
Comment (StadnykYura) [inline src/pages/main/components/SortPanel/SortPanel.tsx]: you don't need the same state here, use parent state for `isDown`

File: src/pages/main/components/SortPanel/SortPanel.tsx [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+  const onChangeDirection = useCallback((isDown: boolean) => {
+    changeDirection(!isDown);
+    onChangeSortingDirection(!isDown);
+  }, []);
```
Comment (StadnykYura) [inline src/pages/main/components/SortPanel/SortPanel.tsx]: because usage of arrow function inside component `onClick` is not a best practice, a possible way to use a callback. But your way is ok.
```
const onChangeDirection = useCallback(() => {
    onChangeSortingDirection(!isDown);
  }, [isDown]);

<Component onClick={onChangeDirection}>

```

File: src/shared/components/ThreeDotsIcon/ThreeDotsIcon.tsx [hunk 12]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import classes from "./ThreeDotsIcon.module.scss";
+
+const ThreeDotsIcon: React.FC = () => (
```
Comment (StadnykYura) [inline src/shared/components/ThreeDotsIcon/ThreeDotsIcon.tsx:4]: i think there is no need for this icon, you can easily do it with css, but it's just a recommendation

File: src/shared/helpers/filterMovies.ts [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+
+export function filterMovies(
+  movies: Movie[],
+  searchingValue: string,
+  categories: string
+) {
+  return movies.filter(
+    (movie) =>
+      (categories === "ALL"
+        ? true
```
Comment (StadnykYura) [inline src/shared/helpers/filterMovies.ts]: make `categories` singular, and it is not good algorithm.
you always go through all list even if `category` is ALL and searchingValue is empty string;
Make it more effective.

File: src/shared/helpers/sortMovies.ts [hunk 14]
```diff
@@ -0,0 +1,9 @@
+import { Movie } from "../../models";
+
+export function sortMovies(movies: Movie[], isDownDirection: boolean) {
+  return movies.sort((movie1, movie2) =>
+    isDownDirection
+      ? movie1.release - movie2.release
+      : movie2.release - movie1.release
+  );
+}
```
Comment (StadnykYura) [inline src/shared/helpers/sortMovies.ts:9]: let it be this way, but on the next HW we need to be able to sort by release or name alphabet, but not down or up. There wasn't information about that.

File: src/pages/main/MainPage.tsx (hunk 2 above)
Comment (StadnykYura) [reply src/pages/main/MainPage.tsx]: also i think, that 
```
const [categories, onChangeCategories] = useState("ALL");
...
const [isDownDirection, onChangeSortingDirection] = useState(true);
 const movies = sortMovies(filterMovies(allMovies, searchingValue, categories), isDownDirection);
```
could be moved to MovieBoard to decrease props drilling.

File: src/pages/main/components/MovieBoard/MovieBoard.tsx (hunk 8 above)
Comment (DmytroMykhailiukAmway) [reply src/pages/main/components/MovieBoard/MovieBoard.tsx]: @StadnykYura I prefer when we can use components even do not pass the props to them and UI doesn't break when we do it. But in this case we can declare component without default props.

File: src/pages/main/components/CategoryPanel/CategoryPanel.tsx (hunk 7 above)
Comment (DmytroMykhailiukAmway) [reply src/pages/main/components/CategoryPanel/CategoryPanel.tsx:14]: @StadnykYura because when i use classes.category-panel i get a lint error

File: src/pages/main/components/CategoryPanel/CategoryPanel.tsx (hunk 6 above)
Comment (DmytroMykhailiukAmway) [reply src/pages/main/components/CategoryPanel/CategoryPanel.tsx]: @StadnykYura I prefer when we can use components even do not pass the props to them and UI doesn't break when we do it. But in this case i could create this component as stateless

File: src/pages/main/components/SearchPanel/SearchPanel.tsx [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+        <Input 
+          value={value} 
+          placeholder="What do you want to watch?"
+          onChange={onChangeValue}
+        />
+      </div>
+      <div className={classes.search__button}>
+        <Button 
+          type={ButtonTypes.PRIMARY} 
+          onButtonClicked={() => onChangeSearchingValue(value)}
```
Comment (StadnykYura) [inline src/pages/main/components/SearchPanel/SearchPanel.tsx:26]: consider to avoid arrow function, it is a bad practice
